ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -449,9 +475,85 @@ public boolean terminateAllQueries() {
     return true;
   }
 
+  public void removeTemporaryQuery(QueryMetadata queryMetadata) {
+    this.allLiveQueries.remove(queryMetadata);
+  }
+
   public DDLCommandResult executeDdlStatement(final DDLStatement statement,
                                               final Map<String, Object> streamsProperties) {
     return queryEngine.handleDdlStatement(statement, streamsProperties);
   }
 
+  private class KsqlEngineMetrics implements Closeable {
```

Comment:
I can see this class getting bigger, it might be worthwhile making in a public class, which also keeps the engine smaller.

Reply:
+1 for me on what @bluemonk3y said.

Reply:
I made this a public class in an 'internal' package.